Core.DomainServices/Repositories/KLE/KLEUpdateHistoryItemRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using Core.DomainModel.KLE;
+using Infrastructure.Services.DataAccess;
+
+namespace Core.DomainServices.Repositories.KLE
+{
+    public class KLEUpdateHistoryItemRepository : IKLEUpdateHistoryItemRepository
+    {
+        private readonly IGenericRepository<KLEUpdateHistoryItem> _updateHistoryItems;
+        private readonly ITransactionManager _transactionManager;
+
+        public KLEUpdateHistoryItemRepository(IGenericRepository<KLEUpdateHistoryItem> updateHistoryItems, ITransactionManager transactionManager)
+        {
+            _updateHistoryItems = updateHistoryItems;
+            _transactionManager = transactionManager;
+        }
+
+        public IEnumerable<KLEUpdateHistoryItem> Get()
+        {
+            return _updateHistoryItems.Get();
+        }
+
+        public KLEUpdateHistoryItem Insert(DateTime version, int userId)
+        {
+            KLEUpdateHistoryItem result;
+            using (var transaction = _transactionManager.Begin(IsolationLevel.Serializable))
+            {
+                result = _updateHistoryItems.Insert(new KLEUpdateHistoryItem(version, userId));
+                _updateHistoryItems.Save();
+                transaction.Commit();
+            }
+            return result;
+        }
+
+        public DateTime GetLastUpdated()
+        {
+            var lastUpdated = DateTime.MinValue;
```

Comment:
@strongmindsmhs  Man kunne også bruge en Maybe<DateTime> for at udtrykke at den var der eller ikke var der

Reply:
@mrjsawdk Hm. Synes følgende er mere verbose og rippler ind i den omkringliggende kode som tvinges til at tage stilling:
        public Maybe<DateTime> GetLastUpdated()
        {
            var previouslyUpdated = (_updateHistoryItems.Count > 0);
            if (previouslyUpdated)
            {
                return Maybe<DateTime>.Some(_updateHistoryItems.GetMax(item => item.Version));
            }
            return Maybe<DateTime>.None;
        }